### PR TITLE
fix: miscellaneous bug fixes

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

re #3270

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Review by dstala — COMMENTED** [review]
all changes verified but for user-menu handling in editor/viewer/commenter roles

File: packages/nc-gui-v2/components/dashboard/TreeView.vue [hunk 1]
```diff
@@ -352,7 +352,7 @@ function openTableCreateDialog() {
                           </div>
                         </a-menu-item>
 
-                        <a-menu-item v-if="isUIAllowed('table-delete')" @click="() => $e('c:table:delete') && deleteTable(table)">
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/dashboard/TreeView.vue:355]: If this gets removed there's no telemetry event sent out as we can't bind the directive onto `a-menu-item`

File: packages/nc-gui-v2/components/dashboard/TreeView.vue [hunk 2]
```diff
@@ -386,10 +386,7 @@ function openTableCreateDialog() {
               </div>
             </a-menu-item>
 
-            <a-menu-item
-              v-if="isUIAllowed('table-delete')"
-              @click="() => $e('c:table:delete') && deleteTable(contextMenuTarget.value)"
-            >
+            <a-menu-item v-if="isUIAllowed('table-delete')" @click="deleteTable(contextMenuTarget.value)">
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/dashboard/TreeView.vue:389]: Same as above

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui-v2/components/dashboard/TreeView.vue (hunk 1 above)
Comment (pranavxc) [reply packages/nc-gui-v2/components/dashboard/TreeView.vue:355]: It's already emitting event from corresponding function in composable

Comment (bcakmakoglu) [reply packages/nc-gui-v2/components/dashboard/TreeView.vue:355]: ah okay I missed that 👍

**Review by bcakmakoglu — APPROVED** [review]

**Comment by pranavxc** [comment]
> c:table:delete

Fixed :+1:

**Comment by pranavxc** [comment]
> all changes verified but for user-menu handling in editor/viewer/commenter roles

Fixed :+1:

**Comment by dstala** [comment]
@pranavxc 
I pushed 2 minor corrections. Review, please. If fine, you can merge. Verified changes from my end.

**Pull request merged**